llvm/lib/Transforms/InstCombine/InstCombineAndOrXor.cpp

```diff
@@ -4593,8 +4593,12 @@ Instruction *InstCombinerImpl::visitXor(BinaryOperator &I) {
   Value *Op0 = I.getOperand(0), *Op1 = I.getOperand(1);
   Value *M;
   if (match(&I, m_c_Xor(m_c_And(m_Not(m_Value(M)), m_Value()),
-                        m_c_And(m_Deferred(M), m_Value()))))
-    return BinaryOperator::CreateDisjointOr(Op0, Op1);
+                        m_c_And(m_Deferred(M), m_Value())))) {
+    if (isGuaranteedNotToBeUndefOrPoison(M))
```

Comment:
```suggestion
    if (isGuaranteedNotToBeUndef(M))
```
It is safe if M is a poison value.

Reply:
Update, thanks @dtcxzyw